python/lsst/ts/love/csc/csc.py

```diff
@@ -0,0 +1,212 @@
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__all__ = ["CSC"]
+
+import asyncio
+import functools
+
+from lsst.ts import salobj
+
+from . import __version__
+from .config_schema import CONFIG_SCHEMA
+from .authorization_model import AuthorizationModel
+from .utils import get_available_components
+
+
+class CSC(salobj.ConfigurableCsc):
+    """Commandable SAL Component for LOVE.
+
+    This CSC is in charge of managing the auth list.
+    """
+
+    valid_simulation_modes: tuple = (0,)
+    version: str = __version__
+
+    def __init__(
+        self,
+        config_dir=None,
+        initial_state=salobj.State.STANDBY,
+        simulation_mode=0,
+        components_to_handle=None,
+    ) -> None:
+        super().__init__(
+            name="LOVE",
+            index=0,
+            config_schema=CONFIG_SCHEMA,
+            config_dir=config_dir,
+            initial_state=initial_state,
+            simulation_mode=simulation_mode,
+        )
+
+        self.config = None
+
+        self.timeout_short = 5.0
+
+        self.authorization_model = AuthorizationModel()
+
+        self._remotes = dict()
+
+        self._create_remotes_for_components(
+            components_to_handle
+            if components_to_handle is not None
+            else (get_available_components() - {"LOVE"})
+        )
+
+    async def end_enable(self, data):
+        await self._set_auth_list_callbacks()
+
+    async def end_disable(self, data):
+        await self._unset_auth_list_callbacks()
+
+    async def do_requestAuthorization(self, data) -> None:
+        """"""
+        self.assert_enabled()
+
+        self.cmd_requestAuthorization.ack_in_progress(
+            data, timeout=self.timeout_request_authorization + self.timeout_short
+        )
+
+        await asyncio.wait_for(
+            self.process_request(data), timeout=self.timeout_request_authorization
+        )
+
+    async def configure(self, config) -> None:
+        """Configure CSC.
+
+        Parameters
+        ----------
+        config : `types.SimpleNamespace`
+        """
+        self.config = config
+        self.authorization_model.set_authorization_mode(
+            auto=self.config.auto_authorization
+        )
```

Comment:
Please raise an exception if the auto parameter is false, since you do not yet handle that case.